src/module_3/Module3_Milestone1.md

```diff
@@ -0,0 +1,1825 @@
+      <td>0</td>
+      <td>0.038462</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>6</td>
+      <td>8</td>
+      <td>0</td>
+      <td>0.038462</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>46</td>
+      <td>44</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>6</td>
+      <td>8</td>
+    </tr>
+    <tr>
+      <th>5</th>
+      <td>0</td>
+      <td>0.038462</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>6</td>
+      <td>10</td>
+      <td>0</td>
+      <td>0.038462</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>46</td>
+      <td>44</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>6</td>
+      <td>10</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+```python
+# Train the Logistic Regression model
+X_train_final_scaled = scaler.fit_transform(X_train_final[reduced_cols])
+X_val_final_scaled = scaler.fit_transform(X_val_final[reduced_cols])
+
+# Prepare the plot environments for both training and validation
+fig_train, axes_train = plt.subplots(1, 2, figsize=(14, 6))
+fig_train.suptitle('Training Data - Performance Curves')
+fig_val, axes_val = plt.subplots(1, 2, figsize=(14, 6))
+fig_val.suptitle('Validation Data - Performance Curves')
+
+c = 1e-6
+lr = LogisticRegression(penalty='l2', C=c)
+
+# Fit the model on scaled training data
+lr.fit(X_train_final_scaled, Y_train)
+
+# Predict probabilities for the positive class on the training set
+train_proba = lr.predict_proba(X_train_final_scaled)[:, 1]
+plot_curves_with_hyperparameter(axes_train, Y_train, train_proba, f'Train C={c}', curve_type="both")
+
+
+# Predict probabilities for the positive class on the validation set
+val_proba = lr.predict_proba(X_val_final_scaled)[:, 1]
+plot_curves_with_hyperparameter(axes_val, Y_val, val_proba, f'Val C={c}', curve_type="both")
+
+# Add legends and show plots
+axes_train[0].legend(loc='upper right')
+axes_train[1].legend(loc='lower right')
+axes_val[0].legend(loc='upper right')
+axes_val[1].legend(loc='lower right')
+
+plt.tight_layout()
+fig_train.show()
+fig_val.show()
+```
+
+    /tmp/ipykernel_82778/1566344662.py:33: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
+      fig_train.show()
+    /tmp/ipykernel_82778/1566344662.py:34: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
+      fig_val.show()
+
+
+
+    
+![png](Module3_Milestone1_files/Module3_Milestone1_60_1.png)
+    
+
+
+
+    
+![png](Module3_Milestone1_files/Module3_Milestone1_60_2.png)
+    
+
+
+We can observe that adding categorical variables does not make the model improve in its performance.
+
+### 6. Evaluating the final model
+
+The final model selected will be the logistic regression with L2 penalty and c=1e-3. Additionally, the model will only be trained with the subset of numerical column previously defined.
+
+
+```python
+lr = LogisticRegression(penalty = 'l2', C = 1e-6)
```

Comment:
The C value is wrong, set to 1e-3 leading to nonsensical results due to too high regularisation